solvers/SE3MotionPlanningTree.m

```diff
@@ -0,0 +1,81 @@
+classdef SE3MotionPlanningTree < CompositeVertexArrayTree
+  properties
+    rbm
+    min_distance = 0.2;
+  end
+
+  methods
+    function obj = SE3MotionPlanningTree()
+      T_R3 = R3MotionPlanningTree();
+      T_SO3 = SO3MotionPlanningTree();
+      obj = obj@CompositeVertexArrayTree({T_R3, T_SO3},{'R3', 'SO3'});
+      obj = obj.setLCMGL('SE3MotionPlanningTree',[0, 0, 0]);
+      urdf = fullfile(getDrakePath, 'systems', 'plants', 'test', 'FallingBrick.urdf');
+      options.floating = true;
+      obj.rbm = RigidBodyManipulator(urdf, options);
+      obj.rbm = obj.rbm.removeCollisionGroupsExcept({});
+      % TODO: Switch to something like the below.
+      %obj.rbm = RigidBodyManipulator();
+      %obj.rbm.name{1} = 'robot';
+      %body = RigidBody();
+      %body = body.setInertial(1, zeros(3,1), eye(3));
+      %body.linkname = 'body';
+      %body.robotnum = 1;
+      %obj.rbm = obj.rbm.addLink(body);
+      %obj.rbm = obj.rbm.addFloatingBase(1, 2, [0;0;0], [0;0;0],'quat'); 
+      %obj.rbm = obj.rbm.addFloatingBase(1, 2, [0;0;0], [0;0;0]); 
+      obj = obj.compile();
+    end
+
+    function obj = compile(obj)
+      obj.rbm = obj.rbm.compile();
+    end
+
+    function valid = checkConstraints(obj, q)
+      valid = checkConstraints@CompositeVertexArrayTree(obj, q);
+      valid = valid && obj.isCollisionFree(q);
+    end
+
+    function valid = isCollisionFree(obj, q)
+      xyz = q(1:3);
+      quat = q(4:7); 
+      rpy = quat2rpy(quat);
+      phi = obj.rbm.collisionDetect([xyz; rpy]);
+      valid = all(phi > obj.min_distance);
+    end
+
+    function obj = addGeometryToRobot(obj, geom)
+      obj.rbm = obj.rbm.addGeometryToBody(2, geom);
+    end
+
+    function obj = addGeometryToWorld(obj, geom)
+      obj.rbm = obj.rbm.addGeometryToBody(1, geom);
+    end
+
+    function obj = setOrientationWeight(obj, orientation_weight)
```

Comment:
I'm drawing this treatment from [Kuffner's paper on sampling and distance metrics in SE(3)](http://www.cs.cmu.edu/afs/cs/academic/class/15494-s14/readings/kuffner_icra2000.pdf).